nix_update/version/github.py

```diff
@@ -34,3 +34,29 @@ def fetch_github_versions(url: ParseResult) -> List[Version]:
     tree = ET.fromstring(resp.read())
     releases = tree.findall(".//{http://www.w3.org/2005/Atom}entry")
     return [version_from_entry(x) for x in releases]
+
+
+def fetch_github_snapshots(url: ParseResult, branch: str) -> List[Version]:
+    if url.netloc != "github.com":
+        return []
+    parts = url.path.split("/")
+    owner, repo = parts[1], parts[2]
+    repo = re.sub(r"\.git$", "", repo)
+    feed_url = f"https://github.com/{owner}/{repo}/commits/{branch}.atom"
+    info(f"fetch {feed_url}")
+    resp = urllib.request.urlopen(feed_url)
+    tree = ET.fromstring(resp.read())
+    commits = tree.findall(".//{http://www.w3.org/2005/Atom}entry")
+
+    for entry in commits:
+        link = entry.find("{http://www.w3.org/2005/Atom}link")
+        updated = entry.find("{http://www.w3.org/2005/Atom}updated")
+        assert (
+            link is not None and updated is not None and updated.text is not None
+        ), "cannot parse ATOM feed"
+        url = urlparse(link.attrib["href"])
+        commit = url.path.rsplit("/", maxsplit=1)[-1]
+        date = updated.text.split("T", maxsplit=1)[0]
+        return [Version(f"unstable-{date}", rev=commit)]
+
+    return []
```

Comment:
Maybe we should just abort here instead.

Reply:
We shouldn't, when more `branch_snapshots_fetchers` fetchers are added, all the fetchers will be called in order and [fetch_latest_version](https://github.com/Mic92/nix-update/blob/56b583e67ca1c06983bcf8fb483afb57127fdd5f/nix_update/version/__init__.py#L97-L99) will throw an error if nothing is found.